src/pytorch_lightning/strategies/launchers/multiprocessing.py

```diff
@@ -209,3 +223,53 @@ class _WorkerOutput(NamedTuple):
     trainer_state: TrainerState
     trainer_results: Any
     extra: _FakeQueue
+
+
+@dataclass
+class _GlobalStateSnapshot:
+    """Captures a hand-selected set of (global) variables in modules and provides a way to restore them.
+
+    It facilitates and encapsulates the transfer of globals like PyTorch's deterministic flags or random generator state
+    across process boundaries when launching processes with :func:`torch.multiprocessing.spawn`.
+
+    Example:
+
+        .. code-block:: python
+
+            # in main process
+            snapshot = _GlobalStateSnapshot.capture()
+
+            # in worker process
+            snapshot.restore()
+    """
+
+    use_deterministic_algorithms: bool
+    use_deterministic_algorithms_warn_only: bool
```

Comment:
maybe capture these 2 in a dict? then you can pass it by checking the `torch_version` only once.

Reply:
Tried but makes it only more complicated, and requires to introduce new names. I prefer the explicit way as it is also easy to annotate with types.
The torch version check can be removed once our minimal version is 1.11.